### DISCUSSION TITLE
Problem with $document->getFullPath() in Chinese when using LinkGenerator for DataObjects

### DISCUSSION THREAD

**Discussion by akno77** [pimcore/pimcore#16761] [Q&A / Support] [answered]
Hi,

I have a problem with a Chinese language site. When I link Chinese pages internally via regular Pimcore links the URLs to the page contain the Chinese special chars. But if I generate links for Data Objects via a link generator the Chinese special chars in the URL are encoded like "%25E5%25B7%25A5%25E8%2589%25BA%25E6%258A%2580%25E6%259C%25AF" instead of "工艺技术".

I did some debugging and found out the problem is in the method "getFullPath" in the Document class. This method calls the method "prepareFrontendPath" at the end which does a "urlencode_ignore_slash".

```
private function prepareFrontendPath(string $path): string
    {
        if (Tool::isFrontend()) {
            $path = urlencode_ignore_slash($path);

            $event = new GenericEvent($this, [
                'frontendPath' => $path,
            ]);
            $this->dispatchEvent($event, FrontendEvents::DOCUMENT_PATH);
            $path = $event->getArgument('frontendPath');
        }

        return $path;
    }
```
```
function urlencode_ignore_slash(string $var): string
{
    $scheme = parse_url($var, PHP_URL_SCHEME);

    if ($scheme) {
        $var = str_replace($scheme . '://', '', $var);
    }

    $placeholder = 'x-X-x-ignore-' . md5(microtime()) . '-slash-x-X-x';
    $var = str_replace('/', $placeholder, $var);
    $var = rawurlencode($var);
    $var = str_replace($placeholder, '/', $var);

    if ($scheme) {
        $var = $scheme . '://' . $var;
    }

    // allow @2x for retina thumbnails, ...
    $var = preg_replace("/%40([\d]+)x\./", '@$1x.', $var);

    return $var;
}
```

Is it already known, that $document->getFullPath() doesn't work correctly with languages like Chinese? The method $document->getPath() doesn't do this special encodings.

Does anyone know why this is done? Is there any solution for getting the same value as for $document->getFullPath() but without the encodings?

**Comment by kingjia90** [comment, marked as answer]
weird, don't know the idea behind it, but seems you would have to do `urldecode(rawurldecode('%25E5%25B7%25A5%25E8%2589%25BA%25E6%258A%2580%25E6%259C%25AF'));` to get back `工艺技术`

**Reply by akno77** [reply]
Thanks, this seems to be a workaround for me!
